modules/login/php/ajaxlogin.class.inc

```diff
@@ -0,0 +1,386 @@
+<?php
+/**
+ * POST request methods.
+ *
+ * Used to retrieve info & login by the login form.
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Login
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\login;
+
+use LORIS\Http\StringStream;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use SinglePointLogin;
+
+/**
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  Login
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class AjaxLogin extends \NDB_Page
+{
+    public $skipTemplate = true;
+    public $AjaxModule   = true;
+
+    /**
+     * This function will return a json object for login.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        /**
+         * Ensure POST request.
+         */
+        if ($request->getMethod() != 'POST') {
+            return (new \LORIS\Http\Response())
+                ->withHeader('Content-Type', 'application/json')
+                ->withStatus(405)
+                ->withHeader('Allow', 'POST')
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        'Only POST requests are allowed'
+                    )
+                );
+        }
+        /**
+         * Parse POST request body.
+         */
+        $requestBody = $request->getParsedBody();
+
+        /**
+         * Response to reply.
+         */
+        $response = $this->processRequest($requestBody);
+
+        return (new \LORIS\Http\Response())
+            ->withHeader('Content-Type', 'application/json')
+            ->withStatus(200)
+            ->withHeader('Allow', 'POST')
+            ->withBody(
+                new StringStream(json_encode($response))
+            );
+    }
+
+    /**
+     * Processes the request values.
+     *
+     * @param array $values the user data received.
+     *
+     * @return array
+     */
+    function processRequest(array $values) : array
+    {
+        $response = array();
```

Comment:
I think you should add the following here:

```php
if (!isset($values['command'])) {
   // do some sort of error handling + early return here
}
```

And below this you can remove all the `isset` checks and proceed knowing that it's good.